### ISSUE TITLE
* cannot be matched in function "Like"

### ISSUE THREAD

**Issue by hellomanjack** [arangodb/arangodb#7471]
Example:the following setence cannot match {"code": "123a*b"}, but it can match {"code":"123aab"}
for entry in firstCollection
filter entry.code like "%123a\*b%"
return entry

![image](https://user-images.githubusercontent.com/25783698/49061906-1bf14380-f24d-11e8-9a88-ff231512a33a.png)
from RegexCache.cpp

**Comment by jsteemann** [comment]
Sorry, I am not sure if I understand the problem.
I just created a collection `f` with the following documents in it:
```
[ 
  { 
    "_key" : "13827406", 
    "_id" : "f/13827406", 
    "_rev" : "_XzREiqy--_", 
    "code" : "123aab" 
  }, 
  { 
    "_key" : "13827415", 
    "_id" : "f/13827415", 
    "_rev" : "_XzREmX6--_", 
    "code" : "123a*b" 
  } 
]
```
And then running the query `FOR doc IN f FILTER doc.code LIKE "%123a*b%" RETURN doc` for me always returns the sought-for document:
```
[ 
  { 
    "_key" : "13827415", 
    "_id" : "f/13827415", 
    "_rev" : "_XzREmX6--_", 
    "code" : "123a*b" 
  } 
]
```
I tried this in 3.3 and 3.4.
Please can you describe exactly what you are trying to achieve and what does not work?
Thanks!

**Comment by hellomanjack** [comment]
@jsteemann yes! It has been fixed on 3.3.19. Well Done.
